Sources/Support/StoreMessageType.swift

```diff
@@ -40,6 +47,14 @@ extension Message.Reason {
         case .priceIncreaseConsent: return .priceIncreaseConsent
         case .generic: return .generic
         default:
+            // winBackOffer message reason was added in iOS 18.0, but it's not recognized by xcode versions <16.0.
+            // https://developer.apple.com/documentation/xcode-release-notes/xcode-14_3-release-notes
+            #if swift(>=6.0)
```

Comment:
See the PR description for more details on why this is necessary

Reply:
the `swift` compilation condition defines the swift mode, whereas `compiler` defined the swift compiler version.

A lot of people will compile in Swift 5.X mode while using Xcode. Since what we actually want to do is detect the Xcode version, and Xcode 16 comes with the Swift 6.0 compiler, it's better to use the `compiler` condition instead of `swift`, as we know if ties directly the Xcode version.

https://docs.swift.org/swift-book/documentation/the-swift-programming-language/statements/#Conditional-Compilation-Block

Reply:
Woah, TIL! Will switch to using the `compiler` condition. Thank you!!

Reply:
Switched in to using the `compiler` condition in https://github.com/RevenueCat/purchases-ios/pull/4343/commits/086dce6a8a23877eeae163aeb0a6c44033409712 😎